### ISSUE TITLE
[4.5.0] Machine settings unreadable

### ISSUE THREAD

**Issue by njibhu** [Ultimaker/Cura#7300]
**Application version**
4.4.0 and 4.5.0 (Did not test earlier)

**Platform**
Archlinux, both from community repo and AppImage.
KDE/Plasma 5.18.3 with default light theme, Xorg.

**Reproduction steps**
1. Open the machine settings

**Screenshot(s)**
![MachineSettings](https://user-images.githubusercontent.com/4581742/77015115-2b315e80-6974-11ea-8007-805988d39cc6.png)

**Actual results**
The settings previously set cannot be read, the text is offset up from the input field.

**Expected results**
Can read the previously entered settings.

**Log file**
```
2020-03-18 23:44:15,620 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/ComboBoxWithOptions.qml:74: TypeError: Cannot call method 'match' of undefined
2020-03-18 23:44:15,623 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/ComboBoxWithOptions.qml:74: TypeError: Cannot call method 'match' of undefined
2020-03-18 23:44:15,626 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/ComboBoxWithOptions.qml:74: TypeError: Cannot call method 'match' of undefined
2020-03-18 23:44:15,629 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/qml/UM/Dialog.qml:86:13: QML Loader: Binding loop detected for property "active"
2020-03-18 23:44:15,632 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/NumericTextFieldWithUnit.qml:43:34: Unable to assign [undefined] to QString
2020-03-18 23:44:15,636 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/NumericTextFieldWithUnit.qml:43:34: Unable to assign [undefined] to QString
2020-03-18 23:44:15,639 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/NumericTextFieldWithUnit.qml:43:34: Unable to assign [undefined] to QString
2020-03-18 23:44:15,642 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/ComboBoxWithOptions.qml:39:34: Unable to assign [undefined] to QString
2020-03-18 23:44:15,645 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/SimpleCheckBox.qml:35:30: Unable to assign [undefined] to QString
2020-03-18 23:44:15,648 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/SimpleCheckBox.qml:35:30: Unable to assign [undefined] to QString
2020-03-18 23:44:15,650 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/SimpleCheckBox.qml:35:30: Unable to assign [undefined] to QString
2020-03-18 23:44:15,651 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/ComboBoxWithOptions.qml:39:34: Unable to assign [undefined] to QString
2020-03-18 23:44:15,653 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/NumericTextFieldWithUnit.qml:43:34: Unable to assign [undefined] to QString
2020-03-18 23:44:15,655 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/PrintHeadMinMaxTextField.qml:56:16: Unable to assign [undefined] to QString
2020-03-18 23:44:15,657 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/PrintHeadMinMaxTextField.qml:38: SyntaxError: JSON.parse: Parse error
2020-03-18 23:44:15,658 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/NumericTextFieldWithUnit.qml:43:34: Unable to assign [undefined] to QString
2020-03-18 23:44:15,660 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/PrintHeadMinMaxTextField.qml:56:16: Unable to assign [undefined] to QString
2020-03-18 23:44:15,662 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/PrintHeadMinMaxTextField.qml:38: SyntaxError: JSON.parse: Parse error
2020-03-18 23:44:15,663 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/NumericTextFieldWithUnit.qml:43:34: Unable to assign [undefined] to QString
2020-03-18 23:44:15,665 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/PrintHeadMinMaxTextField.qml:56:16: Unable to assign [undefined] to QString
2020-03-18 23:44:15,667 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/PrintHeadMinMaxTextField.qml:38: SyntaxError: JSON.parse: Parse error
2020-03-18 23:44:15,668 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/NumericTextFieldWithUnit.qml:43:34: Unable to assign [undefined] to QString
2020-03-18 23:44:15,670 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/PrintHeadMinMaxTextField.qml:56:16: Unable to assign [undefined] to QString
2020-03-18 23:44:15,672 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/PrintHeadMinMaxTextField.qml:38: SyntaxError: JSON.parse: Parse error
2020-03-18 23:44:15,674 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/NumericTextFieldWithUnit.qml:43:34: Unable to assign [undefined] to QString
2020-03-18 23:44:15,675 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/ComboBoxWithOptions.qml:39:34: Unable to assign [undefined] to QString
2020-03-18 23:44:15,677 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/SimpleCheckBox.qml:35:30: Unable to assign [undefined] to QString
2020-03-18 23:44:15,679 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/GcodeTextArea.qml:27:30: Unable to assign [undefined] to QString
2020-03-18 23:44:15,680 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/GcodeTextArea.qml:27:30: Unable to assign [undefined] to QString
2020-03-18 23:44:15,694 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/NumericTextFieldWithUnit.qml:43:34: Unable to assign [undefined] to QString
2020-03-18 23:44:15,696 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/NumericTextFieldWithUnit.qml:43:34: Unable to assign [undefined] to QString
2020-03-18 23:44:15,697 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/NumericTextFieldWithUnit.qml:43:34: Unable to assign [undefined] to QString
2020-03-18 23:44:15,699 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/NumericTextFieldWithUnit.qml:43:34: Unable to assign [undefined] to QString
2020-03-18 23:44:15,701 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/NumericTextFieldWithUnit.qml:43:34: Unable to assign [undefined] to QString
2020-03-18 23:44:15,703 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/GcodeTextArea.qml:27:30: Unable to assign [undefined] to QString
2020-03-18 23:44:15,705 - WARNING - [MainThread] UM.Qt.QtApplication.__onQmlWarning [398]: file:///tmp/.mount_UltimaQnOyFf/usr/bin/resources/qml/MachineSettings/GcodeTextArea.qml:27:30: Unable to assign [undefined] to QString
```

**Additional information**
Change cura theme or QT scaling variable doesn't solve the issue.

**Comment by nallath** [comment]
I don't have the issue myself, so I'm not sure if my changes actually solve it.

**Comment by smartavionics** [comment]
That change made no difference for me although what I see isn't as bad as the example above.

**Comment by Mosher23** [comment]
same on Mac OS X Settings window is also not resisable
<img width="752" alt="Screenshot 2020-03-21 at 11 47 39" src="https://user-images.githubusercontent.com/8256425/77225426-37cdd680-6b6f-11ea-81b7-90581ca4b26b.png">

**Comment by njibhu** [comment]
> I don't have the issue myself, so I'm not sure if my changes actually solve it.

Will try the patch this afternoon and will get back to you



> same on Mac OS X Settings window is also not resisable
> <img alt="Screenshot 2020-03-21 at 11 47 39" width="752" src="https://user-images.githubusercontent.com/8256425/77225426-37cdd680-6b6f-11ea-81b7-90581ca4b26b.png">

I actually can resize the window to make the G-Code input wider, I do not think that the bug is related. Though changing the window size for me doesn't change anything on my bug

**Comment by njibhu** [comment]
@nallath tested by rebuilding an appImage, the problem is still here..

**Comment by njibhu** [comment]
@nallath Found a fix and a way to reproduce it.
Using as a system font Noto Sans 10px, the bug is there. If i switch to Liberation Sans 11pt, it's gone. (with Liberation 10px, the input is still offset, but at least readable).
So it's definitely linked to font size

**Comment by Ghostkeeper** [comment]
Not reproducing the problem with that on Ubuntu when running from source, but it is affected by it. This is my machine settings dialogue for Ender 3 using Noto Sans 10px:

![Screenshot from 2020-03-24 11-12-32](https://user-images.githubusercontent.com/2448634/77413880-6393cb00-6dc0-11ea-941e-b8006356d897.png)

And here is the dialogue using Liberation Sans 11pt:

![image](https://user-images.githubusercontent.com/2448634/77413976-8faf4c00-6dc0-11ea-8363-5fd4602416bd.png)

But when you're using the AppImage, it should always use Noto Sans 10pt.

**Comment by Pimentoso** [comment]
+1 for the problem on MacOS, please make this dialog resizable. Catalina 10.15.4, cura 4.5

**Comment by nallath** [comment]
The dialog has been made resizable already.

**Comment by Mosher23** [comment]
> The dialog has been made resizable already. 

But not on Mac OS or not in 4.5

**Comment by Pimentoso** [comment]
> The dialog has been made resizable already.

Doesn't look like it. Also tried to resize it using a window manager (Rectangle) but it still doesn't change.

**Comment by nallath** [comment]
It wasn't done in 4.5 if I remember correctly.

**Comment by jlsjonas** [comment]
Can this be patched in? @nallath
Or at least a default height fix to make the start & end g-code windows readable on Mac?

**Comment by Ghostkeeper** [comment]
The window is resizable in Cura 4.5 in Linux. If we made a change in 4.6 it must've been to fix something in MacOS only.

I also remember this coming back up recently, but I can't find the commits or any change that we made to fix it, and don't remember that we made any special handling for MacOS then.

**Comment by jlsjonas** [comment]
@rburema might know more? @Ghostkeeper

**Comment by Ghostkeeper** [comment]
This should be fixed in Cura 4.7, through this PR: https://github.com/Ultimaker/Cura/pull/7883
